cpp/splinepy/py/py_spline.hpp

```diff
@@ -373,6 +373,36 @@ class PySpline {
     return cmr;
   }
 
+  /// @brief Evaluate Splines at boundary face centers
+  /// @return numpy array with results
+  py::array_t<double> EvaluateBoundaryCenters() const {
+    // prepare output
+    const int n_faces = 2 * para_dim_;
+    py::array_t<double> queries(n_faces * para_dim_);
+    double* queries_ptr = static_cast<double*>(queries.request().ptr);
+    py::array_t<double> face_centers(n_faces * dim_);
+    double* face_centers_ptr = static_cast<double*>(face_centers.request().ptr);
+
+    for (int i{}; i < para_dim_; i++) {
+      for (int j{}; j < para_dim_; j++) {
+        if (i == j) {
+          queries_ptr[2 * i * para_dim_ + j] = 0.;
+          queries_ptr[(2 * i + 1) * para_dim_ + j] = 1.;
+        } else {
+          queries_ptr[2 * i * para_dim_ + j] = .5;
+          queries_ptr[(2 * i + 1) * para_dim_ + j] = .5;
+        }
+      }
+    }
```

Comment:
This is only valid for bezier. These need to be in relation to `parametric_bounds`

Reply:
good point